main.js

```diff
@@ -0,0 +1,142 @@
+import articleService from "./ArticleService.js";
+import productService from "./ProductService.js";
+
+class Product {
+  constructor(name, description, price, tags, images, favoriteCount = 0) {
+    this.name = name;
+    this.description = description;
+    this.price = price;
+    this.tags = tags;
+    this.images = images;
+    this._favoriteCount = favoriteCount;
+  }
+
+  set favoriteCount(Count) {
+    throw new Error("call favorite()");
+  }
+
+  get favoriteCount() {
+    return this._favoriteCount;
+  }
+
+  favorite() {
+    this._favoriteCount += 1;
+  }
+}
+
+class ElectronicProduct extends Product {
+  constructor(name, description, price, tags, favoriteCount = 0, manufactuer) {
+    super(name, description, price, tags, favoriteCount);
+    this.manufactuer = manufactuer;
+  }
+}
```

Comment:
```suggestion
class ElectronicProduct extends Product {
  constructor(name, description, price, tags, favoriteCount = 0, manufactuer = '') {
    super(name, description, price, tags, favoriteCount);
    this.manufactuer = manufactuer;
  }
}
```
함수 선언 시에 기본값을 지정할 매개변수는 그렇지 않은 매개변수들 보다 뒤에 있어야 합니다. 그렇지 않으면 기본값을 지정하지 않은 매개 변수가 undefined가 되고, 의도하지 않는 동작을 할 수 있게 됩니다.
따라서 위에 예시처럼 기본값을 추가하거나, 매개변수의 순서를 봐꾸면 좋을 것 같습니다.